File: Assignments/Assignment_27/p7.c
#include<stdio.h>

int isDuck(int num){

        int temp=num,count=0,dig,flag=0;
        while(temp>0){

                dig=temp%10;
                if(count==0 && dig==0){

                        return 0;
                }if(count!=0 && dig==0) flag=1;
                temp/=10;
                count++;
        }if(flag==1) return 1;
}

void main(){

        int num;
        printf("Enter a number:\n");
        scanf("%d",&num);
        if(isDuck(num)){

                printf("%d is a duck number\n",num);
        }else{printf("%d is not a duck number\n",num);}
}

File: JAVA/14.Number_System/p7.java
import java.util.*;

class Demo{

	public static void main(String[] args){
	
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter a number:");
		int num=sc.nextInt();
		int temp=num,dig=0,sum=0;
		while(temp>0){
		
			dig=temp%10;
			if(dig%2==0){
				sum+=dig;
			}
			temp/=10;
		}
		System.out.println("sum even digits: "+sum);
	}
}
